src/challenge/Challenge.tsx

```diff
@@ -123,6 +147,12 @@ class Challenge
       this.chContext.actions["fetch-guide"]();
     }
 
+    this.props.bookNode.additionalFiles?.forEach((file) => {
+      if (!(file.filename in this.state.additionalFilesLoaded)) {
+        this.chContext.actions["fetch-file"](file.filename);
```

Comment:
Perhaps the "fetch-file" command could return a promise, so we can just do .then(text => this.setState({
      additionalFilesLoaded: {
        ...this.state.additionalFilesLoaded,
        [filename]: contents,

It would be nice to reduce the number of methods in the interface.